Documentation/gettingstarted/servicemesh/tls-termination.rst

```diff
@@ -0,0 +1,169 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _gs_ingress_tls:
+
+************************************
+Ingress Example With TLS Termination
+************************************
+
+This example builds on the HTTP and gRPC ingress examples, adding TLS
+termination.
+
+Create TLS certificate and private key
+======================================
+
+.. tabs::
+
+    .. group-tab:: Self-signed Certificate
+
+        For demonstration purposes we will use a TLS certificate signed by a made-up,
+        `self-signed <https://cert-manager.io/docs/faq/terminology/#what-does-self-signed-mean-is-my-ca-self-signed>`_
+        certificate authority (CA). One easy way to do this is with `minica <https://github.com/jsha/minica>`_.
+        We want a certificate that will validate ``bookinfo.cilium.rocks`` and
+        ``hipstershop.cilium.rocks``, as these are the host names used in this ingress
+        example.
+
+        .. code-block:: shell-session
+
+            $ minica -domains '*.cilium.rocks'
+
+        On first run, ``minica`` generates a CA certificate and key (``minica.pem`` and
+        ``minica-key.pem``). It also creates a directory called ``_.cilium.rocks``
+        containing a key and certificate file that we will use for the ingress service.
+
+        Create a Kubernetes secret with this demo key and certificate:
+
+        .. code-block:: shell-session
+
+            $ kubectl create secret tls demo-cert --key=_.cilium.rocks/key.pem --cert=_.cilium.rocks/cert.pem
+
+    .. group-tab:: Cert Manager
+
+        Let us install cert-manager:
+
+        .. code-block:: shell-session
+
+            $ helm repo add jetstack https://charts.jetstack.io
+            $ helm install cert-manager jetstack/cert-manager --version v1.7.1 --namespace cert-manager --set installCRDs=true --create-namespace
+
+        Now, create a CA Issuer:
+
+        .. parsed-literal::
+
+            $ kubectl apply -f \ |SCM_WEB|\/examples/kubernetes/servicemesh/ca-issuer.yaml
+
+Deploy the ingress
+==================
+
+The ingress configuration for this demo provides the same routing as those demos
+but with the addition of TLS termination. Deploy this ingress:
+
+
+.. tabs::
+
+    .. group-tab:: Self-signed Certificate
+
+        .. parsed-literal::
+
+            $ kubectl apply -f \ |SCM_WEB|\/examples/kubernetes/servicemesh/tls-ingress.yaml
+
+    .. group-tab:: Cert Manager
+
+        .. parsed-literal::
+
+            $ kubectl apply -f \ |SCM_WEB|\/examples/kubernetes/servicemesh/tls-ingress.yaml
+
+        To tell cert-manager that this Ingress needs a certificate, annotate the
```

Comment:
I'm noticing that you're inconsistent about capitalizing "Ingress" in this section. You should stick to one or the other.

Reply:
thanks, I will go with Ingress instead of ingress.